File: app/src/main/java/com/otraempresa/android/example/RequestTransactionTokenAsyncTask.java
package com.otraempresa.android.example;

import android.content.Context;

import com.kushkipagos.android.Card;
import com.kushkipagos.android.KushkiException;
import com.kushkipagos.android.Transaction;

class RequestTransactionTokenAsyncTask extends AbstractRequestTokenAsyncTask {
    Context appContext;

    RequestTransactionTokenAsyncTask(Context context, String currency) {
        super(context, currency);
        this.appContext=context;
    }


    @Override
    protected Transaction requestToken(Card card) throws KushkiException {
        return kushki.requestToken(card, 19.99,appContext,true);
    }
}

File: app/src/main/java/com/kushkipagos/kushkidemo/services/RequestTransactionTokenAsyncTask.kt
package com.kushkipagos.kushkidemo.services

import android.app.Activity
import android.content.Context
import com.kushkipagos.kushkidemo.contracts.AbstractRequestTokenAsyncTask
import com.kushkipagos.models.Card
import com.kushkipagos.models.Transaction
import kotlinx.coroutines.runBlocking

class RequestTransactionTokenAsyncTask : AbstractRequestTokenAsyncTask {

    val _context: Context
    val _activity: Activity

    constructor(context: Context, activity: Activity) : super(context) {
        _context = context
        _activity = activity
    }

    override fun requestToken(card: Card): Transaction {
        return runBlocking { _kushki.requestToken(card,111111.9, _context, _activity) }
    }

}